### ISSUE TITLE
Automattic\WooCommerce\Admin\API\Options::get_options function is deprecated since version 6.3.    Webhooks issue  403 - 	/v1/billing-agreements/agreement-tokens

### ISSUE THREAD

**Issue by HelpAdude** [woocommerce/woocommerce-paypal-payments#1718]
### Describe the Bug
I'm not a nerd so I apologies  in advance for hopefully explaining an issue here with the plugin.  

on site error:  Automattic\WooCommerce\Admin\API\Options::get_options function is deprecated since version 6.3.

on paypal error 403 /v1/billing-agreements/agreement-tokens

### To Reproduce

1. Go to '…'
2. Click on '…'
3. Scroll down to …'

### Actual Behavior

Webhook is not working nor are payments going through

### Environment

 - WordPress Version  Version 6.3.1 
 - WooCommerce Version Version 8.1.1 |
 - Latest 
 - Browser [e.g. Chrome, Safari] and Version FIREFOX
 - Any other plugins installed yes but plugin was working the day before 

### Additional Details

<details><summary>System status</summary>

  ```
  <!--If applicable, paste the system status here. Please ensure you redact or remove any identifying information. -->
  ```

</details>

**Comment by HelpAdude** [comment]
Disregard Paypal Part :) but the  Automattic\WooCommerce\Admin\API\Options::get_options function is deprecated since version 6.3.           stands as current error

**Comment by InpsydeNiklas** [comment]
Hi @HelpAdude 
Thank you for raising this issue. The warning regarding the `Automattic\WooCommerce\Admin\API\Options::get_options function` being deprecated has been reported in relation other plugins, such as WooPayments. You can reference the similar issue here: https://github.com/Automattic/woocommerce-payments/issues/6849. It's challenging to provide an exact diagnosis without specific details about your website, though.
Usually, warnings like this are displayed when debugging is active. On a live site, debugging should ideally be turned off. Nevertheless, it's good practice to identify and rectify the root cause of any warnings.

As for the 403 error with PayPal, this error likely surfaced because the plugin made repetitive requests for the Reference Transactions status from the PayPal API. Usually, this should only occur once, with the result being cached. However, a bug caused the cache to be invalidated, leading to multiple requests. This has been addressed in the [2.3.0-rc2 version](https://github.com/woocommerce/woocommerce-paypal-payments/releases), slated for official release early next week.

Please note that GitHub is primarily our development platform. For comprehensive assistance, we recommend you contact our [support team directly](https://woocommerce.com/document/woocommerce-paypal-payments/#get-help). If you could provide the support team with more details if you're still facing any issues, it would be greatly appreciated.

We will be closing this GitHub issue for the time being. Once you get in touch with our support team, they will be on hand to help with any issues you might have. We appreciate your understanding and patience!
